Hide short pivot button & Bring back Whitelist feature

YT-Advanced opened the pull request:
#### Hello, I know that you are tired now but I really really need your help to continue this repos :)))
#### As you can see in the code, I have bring back the Whitelist feature by switch from api v1 to api v3. 
## But now I met 2 problems:
- Hide short pivot button didn't work. I have debugged and found the litho is `reel_pivot_button`.
https://github.com/inotia00/ReVanced_Extended/issues/1209

- The whitelist overlay button position is wrong.

### Patches changes: https://github.com/inotia00/revanced-patches/pull/16

YT-Advanced committed:
Add `Hide short pivot button` patch

https://github.com/inotia00/ReVanced_Extended/issues/1209

YT-Advanced committed:
feat(youtube): Re-add Whitelist

Now I'm testing it .-.

YT-Advanced committed:
Update gradle.properties

YT-Advanced committed:
Update release.yml

YT-Advanced committed:
Update PatchStatus.java

YT-Advanced committed:
Update ReVancedSettingsFragment.java

YT-Advanced committed:
Update ReVancedSettingsFragment.java

YT-Advanced committed:
Update ReVancedSettingsFragment.java

YT-Advanced committed:
Update ReVancedSettingsFragment.java

YT-Advanced committed:
Update ReVancedSettingsFragment.java

YT-Advanced committed:
Update ReVancedSettingsFragment.java

YT-Advanced committed:
Update SettingsEnum.java

YT-Advanced committed:
Update WhitelistRequester.java

YT-Advanced committed:
Update SettingsEnum.java

YT-Advanced committed:
Update ResourceHelper.kt

YT-Advanced committed:
Update WhitelistType.java

YT-Advanced committed:
Update ResourceHelper.kt

YT-Advanced committed:
Rename Whitelist.java to Whitelists.java

YT-Advanced committed:
Update package.json

semantic-release-bot committed:
chore(release): 0.115.0 [skip ci]

# [0.115.0](https://github.com/YT-Advanced/revanced-integrations/compare/v0.114.10...v0.115.0) (2023-08-01)

### Features

* **youtube:** preparation for discontinuation ([2449d8c](https://github.com/YT-Advanced/revanced-integrations/commit/2449d8c08509079974ec07d499bc389dfe30e4dd))
* **youtube:** Re-add Whitelist ([930e311](https://github.com/YT-Advanced/revanced-integrations/commit/930e3118f0e47200cdc81b3d86fea37e01d86663))

YT-Advanced commented on app/src/main/java/app/revanced/integrations/settingsmenu/ReVancedSettingsFragment.java:
```diff
@@ -400,6 +408,51 @@ private void setPatchesInformation() {
+                    this.whitelistingPreferenceScreen.addPreference(WhitelistSPEED);
+                }
+            } else {
+                //True is disable
```
@inotia00 I am using `enableDisablePreferences` but look like it is an incorrect code for this case. Here is the old code:
```java
SwitchPreference setWhitelist = (SwitchPreference) findPreferenceOnScreen(SettingsEnum.OVERLAY_BUTTON_WHITELIST.getPath());
this.overlayPreferenceScreen.removePreference(setWhitelist);
this.overlayPreferenceScreen.removePreference(whitelistingPreferenceScreen);

inotia00 commented:
Does youtube api v3 work normally?
As far as I know, you need to get a separate key to access the youtube api v3

YT-Advanced commented:
> Does youtube api v3 work normally?
> As far as I know, you need to get a separate key to access the youtube api v3

Yes, it work. I have about 10 api keys so we can add all of them and don't care about limit. Or we can let user add their key like what Seal are doing.

YT-Advanced committed:
fix(sponsorblock): Auto disable whitelist setting

semantic-release-bot committed:
chore(release): 0.115.1 [skip ci]

## [0.115.1](https://github.com/YT-Advanced/revanced-integrations/compare/v0.115.0...v0.115.1) (2023-08-02)

### Bug Fixes

* **sponsorblock:** Auto disable whitelist setting ([c35b0b4](https://github.com/YT-Advanced/revanced-integrations/commit/c35b0b4fda3e7ac3c611139bdbb37f5cba444855))

inotia00 commented:
If this is not a public API key, it is difficult to merge if the user is issued by the user

because more people are using RVX than you think, and it is impossible to cope when each API key reaches the limits that can be provided

As a typical example, I previously provided the true-worker-ryd API as a mirror server of RYD

but, the developer who provided the API had to close the server in a month because it was used by too many users

Therefore, there is no need to use hard-coded API unless it is a public API key

after discontinued, there is no way for the user to change the hard-coded API

YT-Advanced commented:
@inotia00 I know so I have a fallback method of letting the user get and add the api key themselves. This way we won't need to care about the API limitation. And one more thing, only a small number of users use the whitelist feature to whitelist their favorite channel, so I think user private key (use by default) and 10 API keys from me (in case their key reach rate limit) is enough.

Darkishe commented:
> @inotia00 I know so I have a fallback method of letting the user get and add the api key themselves. This way we won't need to care about the API limitation. And one more thing, only a small number of users use the whitelist feature to whitelist their favorite channel, so I think user private key (use by default) and 10 API keys from me (in case their key reach rate limit) is enough.

Sorry for the off topic but I couldn't find where I could make an issue. Is it possible to add the patch to bypass the age restriction in your fork?

YT-Advanced commented:
> Sorry for the off topic but I couldn't find where I could make an issue. Is it possible to add the patch to bypass the age restriction in your fork?

I will start working on it when this PR finished. But look like it is hard to do.

Darkishe commented:
> > Sorry for the off topic but I couldn't find where I could make an issue. Is it possible to add the patch to bypass the age restriction in your fork?
> 
> I will start working on it when this PR finished. But look like it is hard to do.

This can help you maybe? https://github.com/zerodytrash/Simple-YouTube-Age-Restriction-Bypass

YT-Advanced committed:
fix(connection): 401 Error

YT-Advanced committed:
Update WhitelistRequester.java

YT-Advanced committed:
Done ?

YT-Advanced committed:
Update WhitelistRequester.java

semantic-release-bot committed:
chore(release): 0.115.2 [skip ci]

## [0.115.2](https://github.com/YT-Advanced/revanced-integrations/compare/v0.115.1...v0.115.2) (2023-08-02)

### Bug Fixes

* **connection:** 401 Error ([a42bc70](https://github.com/YT-Advanced/revanced-integrations/commit/a42bc701e77a6a7577f17f7a43f5a72fd27d3f37))

YT-Advanced committed:
Debug

YT-Advanced committed:
Update WhitelistRequester.java

YT-Advanced committed:
Update WhitelistRequester.java

YT-Advanced commented:
@inotia00 Now I have 401 Error :( Can you help me to debug response

YT-Advanced commented:
I think I will close this PR and will open back when I finished
